core/src/main/java/uk/gov/gchq/koryphe/impl/predicate/If.java

```diff
@@ -0,0 +1,213 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.koryphe.impl.predicate;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import uk.gov.gchq.koryphe.predicate.KoryphePredicate;
+import uk.gov.gchq.koryphe.tuple.Tuple;
+import uk.gov.gchq.koryphe.tuple.predicate.IntegerTupleAdaptedPredicate;
+
+import java.util.function.Predicate;
+
+/**
+ * An {@code If} is a {@link Predicate} that conditionally applies one of two predicates to a provided input.
+ *
+ * @param <I> the type of input to be validated
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "class")
+public class If<I> extends KoryphePredicate<I> {
+
+    private Boolean condition;
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "class")
+    private Predicate<? super I> predicate;
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "class")
+    private Predicate<? super I> then;
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "class")
+    private Predicate<? super I> otherwise;
+
+    public If() {
+    }
+
+    public If(final boolean condition, final Predicate<? super I> then) {
+        this(condition, then, null);
+    }
+
+    public If(final boolean condition, final Predicate<? super I> then, final Predicate<? super I> otherwise) {
+        this.condition = condition;
+        this.then = then;
+        this.otherwise = otherwise;
+    }
+
+    public If(final Predicate<? super I> predicate, final Predicate<? super I> then) {
+        this(predicate, then, null);
+    }
+
+    public If(final Predicate<? super I> predicate, final Predicate<? super I> then, final Predicate<? super I> otherwise) {
+        this.predicate = predicate;
+        this.then = then;
+        this.otherwise = otherwise;
+    }
+
+    public Boolean getCondition() {
+        return condition;
+    }
+
+    public void setCondition(final boolean condition) {
+        this.condition = condition;
+    }
+
+    public Predicate<? super I> getThen() {
+        return then;
+    }
+
+    public void setThen(final Predicate<? super I> then) {
+        this.then = then;
+    }
+
+    public Predicate<? super I> getOtherwise() {
+        return otherwise;
+    }
+
+    public void setOtherwise(final Predicate<? super I> otherwise) {
+        this.otherwise = otherwise;
+    }
+
+    public Predicate<? super I> getPredicate() {
+        return predicate;
+    }
+
+    public void setPredicate(final Predicate<? super I> predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public boolean test(final I input) {
```

Comment:
Please can you put this test method just below the constructors. This is more important than the getters and setters and most people who look at this class will want to see this method.